### DISCUSSION TITLE
【建议】支持通过参数远程连接到服务器

### DISCUSSION THREAD

**Discussion by MicroOps-cn** [linuxdeepin/developer-center#2834] [General | 常规讨论]
现在只能通过界面的远程管理，连接到服务器，我是使用的第三方工具(Keepass)来管理密码，现在是使用-e参数执行sshpass命令连接到服务器，但是这样就没法使用上传下载功能。

**Comment by ch593030323** [comment]
没看懂你要表达的意思
deepin-terminal提供了远程的功能，便于远程、上传、下载等

**Comment by MicroOps-cn** [comment]
> 没看懂你要表达的意思 deepin-terminal提供了远程的功能，便于远程、上传、下载等

我现在是使用keepass来管理密码(主要是为了密码安全以及跨电脑\跨平台的同步), keepass可以通过调用命令行来实现快速登陆, 
现在是sshpass来实现绕过输入密码, 但是没法使用右键的上传/下载功能
deepin-terminal 既然支持远程服务器管理,那是否能增加一个直接通过命令行参数来连接服务器
![image](https://user-images.githubusercontent.com/91409141/150713300-f52dd0e1-9ba8-4726-8c03-523a9a1f17d9.png)
![image](https://user-images.githubusercontent.com/91409141/150713499-faf9c5ff-4b5c-4ea3-8f7a-c5deff42ca36.png)

**Comment by ch593030323** [comment]
keepass是一个相当不错的工具
我大概能理解你的想法：在终端用keepass实现远程连接的账号输入
我后面将会把相关建议整理下，和产品反馈
多谢您的建议

**Comment by MicroOps-cn** [comment]
> keepass是一个相当不错的工具 我大概能理解你的想法：在终端用keepass实现远程连接的账号输入 我后面将会把相关建议整理下，和产品反馈 多谢您的建议

谢谢,

**Comment by debuggerx01** [comment]
@MicroOps-cn 其实你只要把`ssh` 替换成 `zssh` 就可以用右键上传下载的功能了

**Reply by MicroOps-cn** [reply]
没法，你是通过调用doConnectServer->doCennectServer->setIsConnectRemote(true)->isConnectRemote()->addAction(tr("Download file"), this, &TermWidget::onDownloadFile)，来实现显示上传下载按钮。

**Reply by MicroOps-cn** [reply]
https://github.com/MicroOps-cn/deepin-terminal/commit/ca3516a4abaf295886392adbab7c466822a4a73f
https://github.com/MicroOps-cn/deepin-terminal/commit/7d5430ce0868f7106a88f8f831aa1803d37ec694
我在我fork的仓库已经实现了该功能
